### ISSUE TITLE
Hotdog not working in KKS

### ISSUE THREAD

**Issue by CrispyBearcon** [Sauceke/hotdog#1]
Hi,

i  am trying to get the hotdog working in KKS. I have enabled it under Lovemachine and restarted the game but its not doing anything.

its outputting stuff in the serial monitor

Do i need to change something with the com port? currently its com18 9600/8/N/1

```
12:59:22.667 -> L0100
12:59:22.667 -> L0100
12:59:22.667 -> L0100
12:59:22.667 -> L0100
12:59:22.667 -> L0100
12:59:22.667 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.715 -> L0100
12:59:22.763 -> L077
12:59:22.763 -> L077
12:59:22.763 -> L077
12:59:22.763 -> L077
12:59:22.763 -> L077
12:59:22.763 -> L075
12:59:22.763 -> L075
12:59:22.811 -> L075
12:59:22.811 -> L075
12:59:22.811 -> L075
12:59:22.811 -> L071
12:59:22.811 -> L071
12:59:22.811 -> L071
12:59:22.811 -> L071
12:59:22.811 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L051
12:59:22.859 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.907 -> L053
12:59:22.955 -> L054
12:59:22.955 -> L054
12:59:22.955 -> L054
12:59:22.955 -> L054
12:59:22.955 -> L055
12:59:22.955 -> L055
12:59:22.955 -> L055
12:59:22.955 -> L055
12:59:23.003 -> L056
12:59:23.003 -> L056
12:59:23.003 -> L056
12:59:23.003 -> L056
12:59:23.003 -> L058
12:59:23.003 -> L058
12:59:23.003 -> L02
12:59:23.003 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.051 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.098 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.146 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.194 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.242 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.290 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.338 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.386 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L00
12:59:23.434 -> L01
12:59:23.434 -> L01
12:59:23.434 -> L04
12:59:23.434 -> L04
12:59:23.481 -> L04
12:59:23.481 -> L04
12:59:23.481 -> L02
12:59:23.481 -> L00
12:59:23.481 -> L00
12:59:23.481 -> L00
12:59:23.481 -> L00
12:59:23.481 -> L00
12:59:23.481 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.529 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.577 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.625 -> L00
12:59:23.673 -> L00
12:59:23.673 -> L00
12:59:23.673 -> L057
12:59:23.673 -> L057
12:59:23.673 -> L057
12:59:23.673 -> L057
12:59:23.673 -> L055
12:59:23.673 -> L055
12:59:23.721 -> L055
12:59:23.721 -> L055
12:59:23.721 -> L053
12:59:23.721 -> L053
12:59:23.721 -> L053
12:59:23.721 -> L053
12:59:23.721 -> L053
12:59:23.721 -> L053
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L09
12:59:23.769 -> L050
12:59:23.817 -> L050
12:59:23.817 -> L09
12:59:23.817 -> L09
12:59:23.817 -> L09
12:59:23.817 -> L09
12:59:23.817 -> L09
12:59:23.817 -> L09
12:59:23.817 -> L08
12:59:23.817 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L08
12:59:23.865 -> L07
12:59:23.865 -> L07
12:59:23.865 -> L07
12:59:23.913 -> L07
12:59:23.913 -> L07
12:59:23.913 -> L07
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.913 -> L06
12:59:23.961 -> L064
12:59:23.961 -> L064
12:59:23.961 -> L06
12:59:23.961 -> L06
12:59:23.961 -> L06
12:59:23.961 -> L06
12:59:23.961 -> L06
12:59:23.961 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.009 -> L06
12:59:24.057 -> L05
12:59:24.057 -> L05
12:59:24.057 -> L05
12:59:24.057 -> L05
12:59:24.057 -> L05
12:59:24.057 -> L090
12:59:24.057 -> L090
12:59:24.057 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L090
12:59:24.105 -> L088
12:59:24.152 -> L088
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.152 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.200 -> L05
12:59:24.247 -> L090
12:59:24.247 -> L090
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.247 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.295 -> L05
12:59:24.343 -> L05
12:59:24.343 -> L05
12:59:24.343 -> L05
12:59:24.343 -> L05
12:59:24.343 -> L05
12:59:24.343 -> L090
12:59:24.343 -> L090
12:59:24.343 -> L090
12:59:24.391 -> L090
12:59:24.391 -> L090
12:59:24.391 -> L091
12:59:24.391 -> L091


```

**Comment by Sauceke** [comment]
Hi, the v0.2 firmware is not supported in LoveMachine at the moment, you have to upload the [v0.1 sketch](https://github.com/Sauceke/hotdog/releases/download/v0.1/hotdog-fw.ino).
On a side note, please allow at least 3 seconds for the device to calibrate after turning it on. From these logs it looks like your R1 and R2 are either faulty or something was blocking them during startup.

**Comment by CrispyBearcon** [comment]
Hi. i flashed the v0.1 sketch, reconnected and waited 20secs before starting the game. still not working. could it maybe be the nano every i use?

**Comment by Sauceke** [comment]
Yes, the Every uses DTR signals and the current impl doesn't support that. Good news is I'll soon release 2.10 which will support both BLE and Every through serial.
Meanwhile you could try this workaround:
https://forum.arduino.cc/t/how-to-prevent-nano-every-from-resetting-upon-connection-to-the-serial-port/674505/2

**Comment by CrispyBearcon** [comment]
setting up the board.local.txt seems not to work. i will wait for the new release :) Thanks!

**Comment by Sauceke** [comment]
2.10 is out, you should be able to connect a nano every + v0.2 firmware through the HotdogServer.

**Comment by Sauceke** [comment]
Does that solve your issue?

**Comment by Sauceke** [comment]
Assuming issue to be resolved.

**Comment by imfrea** [comment]
Hello, I'm experiencing a similar issue.
I built a hotdog with NRF2401 module using a Universal PCB, and flashed the firmware from https://github.com/Sauceke/hotdog/releases/latest/download/hotdog-fw.ino.

After resetting the Arduino Nano and letting it sit for a while, I started HotdogServer.exe and then launched the game. The LoveMachine plugin loads successfully (here's my BepInEx log).
https://pastebin.com/3qwJSCXd

The hotdog server console shows:
`Hotdog server running on port 5365.  
Client connected  
Listening to Hotdog serial device on port COM12`

However, in-game character movements do not respond to the device - nothing moves.
When checking serial messages with UartAssist.exe, it mostly fluctuates between L020 and L0100.
`[2025-05-01 20:50:05.851]# RECV ASCII>
0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
...lots of L0100 when photoresistor is clear...
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L0100
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
...lots of L020 when photoresistor is obstructed...
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
L0

[2025-05-01 20:50:05.935]# RECV ASCII>
20
L020
L020
L020
L020
L020
L020
L020
L020
L020
L020
`

Thank you for your help!
